### PR TITLE
P&P: SSO for Regular Web Apps

### PR DESCRIPTION
**Pull request by mpaktiti**

### CODE REVIEW

**Comment by auth0build** [comment]
Can one of the admins verify this patch?

**Comment by mpaktiti** [comment]
Maria:
- [ ] We are missing the `how to configure a client` part
- [ ] Replace flow images with new ones from design team:
  - authz-code-flow.png
  - login-flow.png
  - logout-flow.png
  - solution-diagram.png
- [x] Add content on home realm discovery
- [x] Add content on Kerberos
- [x] Update Logout and Access Control sections (add more content on theory)

Jerrie:
- [ ] `Inside the Implementation > User Login` is missing the `ASP .NET Core: Implement the Login` sub-paragraph

To be discussed:
- [ ] Format for TOC (I added a temporary one)
- [ ] Add call-outs for additional info/links/videos

File: articles/architecture-scenarios/application/web-app-sso.md [hunk 1]
```diff
@@ -0,0 +1,523 @@
...
+
+There are a number of different ways in which you can implement a Login screen using a web application and Auth0:
+- __Hosted Lock__: Use an instance of the Lock widget which is hosted on the Auth0 infrastructure.
+- __Embedded Lock__: Embed the Lock widget inside a web page of your application. You have some customization options for the actual Lock widget, and full control over the rest of the HTML on the page.
+- __Custom UI__: Develop a completely custom web page for the login screen. The custom HTML form will post back to your server which will in turn authenticate the user using the Authentication API. For more information on when to use a Custom UI refer to [Lock vs. a Custom UI](/libraries/when-to-use-lock).
+
+The recommended best practice is to use Hosted Lock because it is the most secure option and the easiest way to enable users to log in to your application.
+
+#### Automate Home Realm Discovery (HRD)
```
Comment (mpaktiti) [inline articles/architecture-scenarios/application/web-app-sso.md]: @ntotten New paragraph on home realm discovery (review + add comments please)

File: articles/architecture-scenarios/application/web-app-sso.md [hunk 2]
```diff
@@ -0,0 +1,410 @@
...
+
+In this case the developer can pick one of the following solution to this problem:
+
+- The web app falls back on the Auth0 SSO session when it wants to refresh the `id_token` by performing another authorization flow (`/authorize` endpoint) with Auth0.
+
+- When performing the initial authorization flow, the web app requests a `refresh_token`, by adding `offline_access` at the `scope` parameter, for example `scope=openid offline_access`. The `refresh_token` is stored in session, alongside with the `id_token`. Then when a session needs to be refreshed (a timeframe has passed or the user tries to approve a timesheet), the app uses the `refresh_token` on the backend to obtain a new `id_token`, using the `/oauth/token` endpoint with `grant_type=refresh_token`. This will result in re-querying the Active Directory, get the updated information, which would include any new groups the user has been added to since the last time, and sending this information to the web app as part of the `id_token` claims.
```
Comment (mpaktiti) [inline articles/architecture-scenarios/application/web-app-sso.md]: This needs to be tested against pipeline2 https://auth0.slack.com/archives/patterns-practices/p1475506975000342

Comment (mpaktiti) [reply articles/architecture-scenarios/application/web-app-sso.md]: Consider adding 3rd option using `/tokeninfo` (@jerriep to test whether this refreshes the AD permissions)

**Pull request merged**